src/intellisense/texDocSymbolProvider.ts

```diff
@@ -97,27 +98,26 @@ function argContentToStr(argContent: Ast.Node[], preserveCurlyBrace: boolean = f
     * @return: the tree-like TeXElement[] structure
 */
 // reference: https://github.com/James-Yu/LaTeX-Workshop/blob/master/src/outline/structurelib/latex.ts#L30
-export async function genTexElements(document: vscode.TextDocument): Promise<TeXElement[]> {
-    const filePath = document.fileName;
-    if (filePath === undefined) {
-        return [];
-    }
-    const rootElement = { children: [] };
-    let ast = await unifiedParser.parse(document.getText());
+export async function genTexElements(filePath:string, documentText: string, projectCache: ProjectCache): Promise<TeXElement[]> {
+    const resElement = { children: [] };
+    const fileElements = { subFile: [], bibFiles: [] };
+    let ast = unifiedParser.parse(documentText);
     for (const node of ast.content) {
         if (['string', 'parbreak', 'whitespace'].includes(node.type)) {
             continue;
         }
         try {
-            await parseNode(node, rootElement);
+            await parseNode(node, resElement , fileElements);
         }
         catch (e) {
             console.log(e);
         }
     }
-    let struct = rootElement.children as TeXElement[];
+    let struct = resElement.children as TeXElement[];
     struct = hierarchyStructFormat(struct);
-    return struct;
+    projectCache.getSymbolCache()?.set(filePath, struct);
```

Comment:
what is `projectCache`

Reply:
`ProjectCache` is a class used to store the symbols and bibFiles extract from the documents. It mainly maintains two dictionary which mapping the root file to the symbols and bibFiles dict with key as each used file. `ProjectCache` offers an tracking to the bibFile name that used in the project.

Reply:
then please rename the `ProjectCache` with a more reasonable name.

Reply:
Ok, how about 'SymbolBibCache' ?

Reply:
I do not care, but `ProjectCache` is unacceptable